cmake/modules/BuildOpentelemetry.cmake

```diff
@@ -41,8 +39,11 @@ function(build_opentelemetry)
                  ${opentelemetry_cpp_targets})
   endif()
 
-  if(NOT WITH_SYSTEM_BOOST)
+  if(WITH_SYSTEM_BOOST)
+    list(APPEND opentelemetry_CMAKE_ARGS -DBOOST_ROOT=/opt/ceph)
```

Comment:
i ran into similar issues with boost in my arrow submodule work in https://github.com/ceph/ceph/pull/44696

`/opt/ceph` is what some of the builders use, but we shouldn't hard-code that here. you can use `-DBOOST_ROOT=${BOOST_ROOT}` instead to forward ceph's cmake variable into external project

Reply:
thanks!
currently, `${BOOST_ROOT}` is defined only in your PR, right?

Reply:
On not all platforms is there an `/opt/ceph` so flexibility here would be appreciated